### PR TITLE
apps: Fix bug by calling getrandom

### PR DESCRIPTION
**Pull request by Weizihan**
Signed-off-by: weizihan <weizihan@xiaomi.com>

## Summary
fsutils/mkgpt/mkgpt.c
Fix bug by calling getrandom

## Impact
mkgpt.c  calling getrandom 

## Testing
passed vela CI

### CODE REVIEW

**Review by davids5 — CHANGES_REQUESTED** [review]

File: fsutils/mkgpt/mkgpt.c [hunk 1]
```diff
@@ -128,12 +129,12 @@ static const uint8_t g_partition_type_swap[16] =
 
 static void get_uuid(FAR uint8_t *uuid)
 {
-  int fd;
-  fd = open("/dev/urandom", O_RDONLY);
-  if (fd > 0)
+  /* call getrandom to read  /dev/urandom */
```
[the remark this answers is not in the document]
Comment (Weizihan) [reply fsutils/mkgpt/mkgpt.c:132]: Yes

File: fsutils/mkgpt/mkgpt.c [hunk 2]
```diff
@@ -128,12 +129,12 @@ static const uint8_t g_partition_type_swap[16] =
...
+
+  if (getrandom(uuid, 16, 1) < 0)
```
Comment (pkarashchenko) [inline fsutils/mkgpt/mkgpt.c:134]:
```suggestion
  if (getrandom(uuid, 16, GRND_NONBLOCK) < 0)
```

Comment (pkarashchenko) [reply fsutils/mkgpt/mkgpt.c:134]: But current code is different from the original one that was a blocking read.

File: fsutils/mkgpt/mkgpt.c [hunk 3]
```diff
@@ -128,12 +129,12 @@ static const uint8_t g_partition_type_swap[16] =
 
 static void get_uuid(FAR uint8_t *uuid)
 {
-  int fd;
-  fd = open("/dev/urandom", O_RDONLY);
-  if (fd > 0)
+  /* call getrandom to read  /dev/urandom */
+
+  if (getrandom(uuid, 16, GRND_NONBLOCK) < 0)
```
Comment (xiaoxiang781216) [inline fsutils/mkgpt/mkgpt.c:134]:
```suggestion
  if (getrandom(uuid, 16, 0) < 0)
```

Comment (pkarashchenko) [reply fsutils/mkgpt/mkgpt.c:134]: Yes. This gets back to original behaviour

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**